explorer/common/trace_stream.h

```diff
@@ -100,6 +100,18 @@ class TraceStream {
 
   auto current_phase() const -> ProgramPhase { return current_phase_; }
 
+  template <typename U, typename... T>
+  auto log(U&& message, T&&... messages) const -> llvm::raw_ostream& {
+    CARBON_CHECK(is_enabled());
+    if constexpr (sizeof...(messages) == 0) {
+      **stream_ << message;
+      return **stream_ << "\n";
+    } else {
+      **stream_ << message << "\t\t";
+      return log(std::forward<T>(messages)...);
+    }
+  }
```

Comment:
How to make this non-recursive:

```suggestion
  template <typename U, typename... T>
  auto log(U&& message, T&&... messages) const -> llvm::raw_ostream& {
    CARBON_CHECK(is_enabled());
    **stream_ << message;
    auto log_subsequent = [&] (auto&& next) {
      **stream_ << "\t\t" << next;
    };
    (log_subsequent(std::forward<T>(messages)), ...);
    return **stream_ << "\n";
  }
```